C error with `map[u8][3]u8` (value as fixed array)

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.10 ded45dc, press to see full `v doctor` output</summary>

|V full version      |V 0.4.10 8683e63.ded45dc
|:-------------------|:-------------------
|OS                  |linux, Ubuntu 22.04.5 LTS
|Processor           |4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-4150 CPU @ 3.50GHz
|Memory              |0.29GB/3.7GB
|                    |
|V executable        |/home/jorge/v/v
|V last modified time|2025-05-14 22:54:21
|                    |
|V home dir          |OK, value: /home/jorge/v
|VMODULES            |OK, value: /home/jorge/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/jorge/bugs
|                    |
|Git version         |git version 2.34.1
|V git status        |weekly.2025.16-149-gded45dce
|.git/config present |true
|                    |
|cc version          |cc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
|gcc version         |gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
|clang version       |Ubuntu clang version 14.0.0-1ubuntu1.1
|tcc version         |tcc version 0.9.28rc 2024-07-31 HEAD@1cee0908 (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 0134e9b9
|emcc version        |emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.5 ()
|glibc version       |ldd (Ubuntu GLIBC 2.35-0ubuntu3.9) 2.35

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  2025_04_14.v && 2025_04_14`
```v
struct Serials {
	a u8
	b u8
	c u8
pub mut:
	ids map[u8][3]u8
}

pub fn (mut s Serials) id(id u8) {
	if _ := s.ids[id] { 

	} else { 
		s3 := u8(s.c + s.ids.len)
		s.ids[id] = [ s.a, s.a, s3 ]!
	}
}

fn main() {
	mut s := Serials{ a:1 b:2 }
	s.id(u8(3))
}
```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/2025_04_14.01JV8GKQJ87MXQZYZ7R6W734V8.tmp.c:6716: error: lvalue expected
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```
```
$ v -show-c-output run 2025_04_14.v
======== Output of the C Compiler (/home/jorge/v/thirdparty/tcc/tcc.exe) ========
/tmp/v_1000/2025_04_14.01JV8G3V995ZQH2434K87SEX00.tmp.c:6716: error: lvalue expected
=================================================================================
======== Output of the C Compiler (cc) ========
/tmp/v_1000/2025_04_14.01JV8G3V995ZQH2434K87SEX00.tmp.c: In function ‘main__Serials_id’:
/tmp/v_1000/2025_04_14.01JV8G3V995ZQH2434K87SEX00.tmp.c:6716:49: error: assignment to expression with array type
 6716 |                 *((Array_fixed_u8_3*)&_t1.data) = *((Array_fixed_u8_3*)_t2);
      |                                                 ^
/tmp/v_1000/2025_04_14.01JV8G3V995ZQH2434K87SEX00.tmp.c:6722:45: error: invalid initializer
 6722 |                 Array_fixed_u8_3 _dummy_1 = (*(Array_fixed_u8_3*)_t1.data);
      |                                             ^
===============================================
```
**What did you expect to see?**

no error

**Workaround**

Use not fixed array

```v

-	ids map[u8][3]u8
+	ids map[u8][]u8

-		s.ids[id] = [ s.a, s.a, s3 ]!
+		s.ids[id] = [ s.a, s.a, s3 ]
```


> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/if.v
+++ b/vlib/v/gen/c/if.v
@@ -343,9 +343,16 @@ fn (mut g Gen) if_expr(node ast.IfExpr) {
 					} else {
 						branch.cond.vars[0].name
 					}
-					g.write('\t${base_type} ${cond_var_name} = ')
-					g.expr(branch.cond.expr)
-					g.writeln(';')
+					if g.table.sym(branch.cond.expr_type).kind == .array_fixed {
+						g.writeln('\t${base_type} ${cond_var_name} = {0};')
+						g.write('\tmemcpy((${base_type}*)${cond_var_name}, &')
+						g.expr(branch.cond.expr)
+						g.writeln(', sizeof(${base_type}));')
+					} else {
+						g.write('\t${base_type} ${cond_var_name} = ')
+						g.expr(branch.cond.expr)
+						g.writeln(';')
+					}
 				} else {
 					mut is_auto_heap := false
 					if branch.stmts.len > 0 {

--- a/vlib/v/gen/c/index.v
+++ b/vlib/v/gen/c/index.v
@@ -546,7 +546,11 @@ fn (mut g Gen) index_of_map(node ast.IndexExpr, sym ast.TypeSymbol) {
 			opt_val_type := g.styp(val_type.set_flag(.option))
 			g.writeln('${opt_val_type} ${tmp_opt} = {0};')
 			g.writeln('if (${tmp_opt_ptr}) {')
-			g.writeln('\t*((${val_type_str}*)&${tmp_opt}.data) = *((${val_type_str}*)${tmp_opt_ptr});')
+			if val_sym.kind == .array_fixed {
+				g.writeln('\tmemcpy((${val_type_str}*)${tmp_opt}.data, (${val_type_str}*)${tmp_opt_ptr}, sizeof(${val_type_str}));')
+			} else {
+				g.writeln('\t*((${val_type_str}*)&${tmp_opt}.data) = *((${val_type_str}*)${tmp_opt_ptr});')
+			}
 			g.writeln('} else {')
 			g.writeln('\t${tmp_opt}.state = 2; ${tmp_opt}.err = _v_error(_SLIT("map key does not exist"));')
 			g.writeln('}')

--- a/vlib/v/tests/builtin_maps/map_fixed_array_if_guard_test.v
+++ b/vlib/v/tests/builtin_maps/map_fixed_array_if_guard_test.v
@@ -1,0 +1,29 @@
+struct Serials {
+	a u8
+	b u8
+	c u8
+pub mut:
+	ids map[u8][3]u8
+}
+
+pub fn (mut s Serials) id(id u8) {
+	if _ := s.ids[id] {
+	} else {
+		s3 := u8(s.c + s.ids.len)
+		s.ids[id] = [s.a, s.a, s3]!
+	}
+}
+
+fn test_map_fixed_array_if_guard() {
+	mut s := Serials{
+		a: 1
+		b: 2
+	}
+	s.id(u8(3))
+	println(s)
+	assert s.ids[3] == [u8(1), 1, 0]!
+
+	s.id(u8(4))
+	println(s)
+	assert s.ids[4] == [u8(1), 1, 1]!
+}

